corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -816,6 +817,12 @@ def _get_case_and_ledger_updates(domain, sql_form):
     )
 
 
+def _get_case_ids(xform):
+    case_ids = get_case_ids_from_form(xform)
+    case_ids.update(get_case_ids_from_stock_transactions(xform))
```

Comment:
how is this resolving the duplicate key issue?

Reply:
I think the issue was that the migration was attempting to process 2 forms in simultaneously which is what was causing the duplicate key issue. This change adds case IDs referenced by stock transactions into the list of IDs what are used to determine if it's safe to process a form (if no other form already being processed touches any of these IDs)

Reply:
Looking at the other places where `get_case_ids_from_form` is called, most also call `get_case_ids_from_stock_transactions` and the places that don't likely should. Perhaps worth making a function that combines the 2?

Reply:
Good idea. I'll do that in a separate PR.

Reply:
https://github.com/dimagi/commcare-hq/pull/24431